packages/x-date-pickers-pro/src/internal/models/fields.ts

```diff
@@ -16,35 +9,43 @@ export interface RangeFieldSection extends FieldSection {
 export interface RangeFieldSectionWithoutPosition
   extends Omit<RangeFieldSection, 'start' | 'end' | 'startInInput' | 'endInInput'> {}
 
-type BaseMultiInputFieldSlotsComponent = {
-  Root?: React.ElementType<StackProps>;
-  TextField?: React.ElementType<TextFieldProps>;
-  Separator?: React.ElementType<TypographyProps>;
-};
+/**
+ * Props the `textField` slot of the multi input field can receive when used inside a picker.
+ */
+export interface MultiInputFieldSlotTextFieldProps {
+  inputRef?: React.Ref<HTMLInputElement>;
+  disabled?: boolean;
+  readOnly?: boolean;
+  id?: string;
+  label?: React.ReactNode;
+  onKeyDown?: React.KeyboardEventHandler;
+  onFocus?: React.FocusEventHandler;
+  focused?: boolean;
+}
 
-type BaseMultiInputFieldSlotsComponentsProps = {
-  root?: SlotComponentProps<typeof Stack, {}, Record<string, any>>;
-  textField?: SlotComponentProps<
-    typeof TextField,
-    {},
-    { position?: 'start' | 'end' } & Record<string, any>
-  >;
-  separator?: SlotComponentProps<typeof Typography, {}, Record<string, any>>;
-};
+/**
+ * Props the `root` slot of the multi input field can receive when used inside a picker.
+ */
+export interface MultiInputFieldSlotRootProps {
+  onBlur?: React.FocusEventHandler;
+}
 
-export interface BaseMultiInputFieldProps<TValue, TError>
-  extends Omit<
-    BaseFieldProps<TValue, TError>,
-    'components' | 'componentsProps' | 'slots' | 'slotProps'
-  > {
-  /**
-   * @deprecated Please use `slots`.
-   */
-  components?: BaseMultiInputFieldSlotsComponent;
-  /**
-   * @deprecated Please use `slotProps`.
-   */
-  componentsProps?: BaseMultiInputFieldSlotsComponentsProps;
-  slots?: UncapitalizeObjectKeys<BaseMultiInputFieldSlotsComponent>;
-  slotProps?: BaseMultiInputFieldSlotsComponentsProps;
+/**
+ * Props the multi input field can receive when used inside a picker.
+ * Only contains what the MUI component are passing to the field, not what users can pass using the `props.slotProps.field`.
```

Comment:
It was `MUI` in opposition to `the user`

Basically, what I mean is that it only contains stuff we are aware of, if they are passing some additional props through slots, it will not be pass of this typing otherwise we loose the advantage of this interface.

If we replace by `the picker` we loose this bit of information, the picker is passing everything the users gives to him.

Maybe we can just drop the 1st part:

> Do not contain what users pass using the `props.slotProps.field` prop.